File: csharp-reference/Characters/Goku.cs
using DragonBallArena.Models;

namespace DragonBallArena.Characters;

public sealed class Goku(string? name = null) : BaseFighter(name)
{
    public override string Id => "goku";
    public override string DisplayName => "Goku";
    public override string Greeting => "Let's make this a good fight!";
    public override string SignatureAttack => "Kamehameha";
    public override string AttackStyle => "beam";
    public override double Scale => 1.0;
    public override FighterStats Stats => new(9000, 92, 72, 0.18, 19);
    public override FighterPalette Palette => new(
        "#f2b07b", "#17171d", "#f47b20", "#173f8a", "#f4d34e", "#46d9ff");
}

File: src/game/characters/Goku.ts
import { BaseFighter } from "../BaseFighter";

export class Goku extends BaseFighter {
  readonly id = "goku" as const;
  readonly name = "Goku";
  readonly greeting = "Let's make this a good fight!";
  readonly signatureAttack = "Kamehameha";
  readonly attackStyle = "beam" as const;
  readonly power = 9000;
  readonly speed = 92;
  readonly defense = 72;
  readonly criticalChance = 0.18;
  readonly baseDamage = 19;
  readonly nativeFacing = "right" as const;
  readonly scale = 1;
  readonly palette = {
    skin: "#f2b07b",
    hair: "#17171d",
    primary: "#f47b20",
    secondary: "#173f8a",
    accent: "#f4d34e",
    aura: "#46d9ff"
  };
}
